packages/backend-data/API.md

```diff
@@ -40,7 +39,7 @@ export type DataSchema = string | DerivedModelSchema;
 export type DefaultAuthorizationMode = 'iam' | 'userPool' | 'oidc' | 'apiKey' | 'lambda';
 
 // @public
-export const defineData: (props: DataProps) => ConstructFactory<ResourceProvider<AmplifyDataResources>>;
+export const defineData: (props: DataProps) => ConstructFactory<AmplifyData>;

```

Comment:
Same here, AmplifyData implements ResourceProvider<AmplifyDataResources>. Using the AmplifyData type instead of the ResourceProvider type ensures that all of the class methods show up in autocomplete in the backend file.